internal/cloud/backend.go

```diff
@@ -178,18 +183,26 @@ func (b *Cloud) PrepareConfig(obj cty.Value) (cty.Value, tfdiags.Diagnostics) {
 	}
 
 	WorkspaceMapping := WorkspaceMapping{}
+
+	// Initially set the project via env var
+	WorkspaceMapping.Project = os.Getenv("TF_CLOUD_PROJECT")
```

Comment:
I don't think we need line 188 and line 197 to line 199.  They are adding attributes to a temporary WorkspaceMapping object, which get created to call the `Strategy` method (line 195). But the Strategy method is not validating against a wm.Project